### PR TITLE
asserts: add cross checks for snap asserts

### PR DESCRIPTION
**Pull request by pedronis**

### CODE REVIEW

**Comment by pedronis** [comment]
based on #1515

File: asserts/snap_asserts_test.go [hunk 1]
```diff
@@ -128,6 +128,72 @@ func (sds *snapDeclSuite) TestDecodeInvalid(c *C) {
...
+		"timestamp":    time.Now().Format(time.RFC3339),
+	}
+	snapDecl, err := otherDB.Sign(asserts.SnapDeclarationType, headers, nil, "")
+	c.Assert(err, IsNil)
+
+	err = db.Check(snapDecl)
+	c.Assert(err, ErrorMatches, `snap-declaration assertion for "foo" \(id "snap-id-1"\) is not signed by a directly trusted authority:.*`)
+}
+
+func (sds *snapDeclSuite) TestSnapDeclarationCheckMissingAccount(c *C) {
```
Comment (emgee) [inline asserts/snap_asserts_test.go]: Maybe "MissingPublisherAccount" or "MissingPublisher"?

**Comment by emgee** [comment]
+1, with small test name suggestion

**Comment by pedronis** [comment]
retest this please

**Pull request merged**